### ISSUE TITLE
modifier-name-case will need to be updated for First-Class Component Templates

### ISSUE THREAD

**Issue by chriskrycho** [ember-template-lint/ember-template-lint#2436]
This rule makes good sense in a resolver-based world. However, it will *not* make sense (and in fact will be impossible to follow!) in the world of [First-Class Component Templates](http://emberjs.github.io/rfcs/0779-first-class-component-templates.html), where we will be importing or defining modifiers by name as JS identifiers—and dasherized names are not valid JS identifiers!

Borrowing an example from the guides that might be written "inline":

```js
import Component from '@glimmer/component';
import { tracked } from '@glimmer/tracking';
import { modifier } from 'ember-modifier';

const playWhen = modifier((el, [shouldPlay]) => {
  if (shouldPlay) {
    el.play();
  } else {
    el.pause();
  }
});

export default class AudioPlayer extends Component {
  @tracked isPlaying = false;

  <template>
    <audio src={{@srcURL}} {{playWhen this.isPlaying}} />

    <button type='button' {{on "click" this.play}}>Play</button>
    <button type='button' {{on "click" this.pause}}>Pause</button>
  </template>
}
```

Notice that it's *impossible* to name `playWhen` with dash case here. The only way we even *could* do it would be to stamp it onto the component instance like this (and then we'd need to use `get` with it! 😬):

```js
export default class AudioPlayer extends Component {
  // ...

  get 'play-when'() {
    return playWhen;
  }

  <template>
    <audio src={{@srcURL}} {{(get this 'play-when') this.isPlaying}} />

    <button type='button' {{on "click" this.play}}>Play</button>
    <button type='button' {{on "click" this.pause}}>Pause</button>
  </template>
}
```

This is needless overhead and also just extremely weird!

**Net:** this rule should be turned *off* in the context of strict-mode templates, and it should probably be reevaluated even for loose-mode templates now that modifiers can be passed as arguments and yielded using the `{{modifier}}` helper (post-3.25).
